### PR TITLE
Add software version type to system info

### PR DESCRIPTION
**Pull request by CY-1992**
I do not have complete understanding of MAVSDK package. I tried to implement version type in system info.  I hope this pull request would close [#1910 ](https://github.com/mavlink/MAVSDK/issues/1910). Also created a pull request [#298](https://github.com/mavlink/MAVSDK-Proto/pull/298) in MAVSDK-Proto

### CODE REVIEW

File: examples/system_info/system_info.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
+//
+// Simple example to demonstrate how to query system info using MAVSDK.
```
Comment (julianoes) [inline examples/system_info/system_info.cpp:2]: Awesome, thanks for adding this.

File: examples/system_info/system_info.cpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    auto info = Info{system};
+
+    // Wait until version/firmware information has been populated from the vehicle
+    while (info.get_identification().first == Info::Result::InformationNotReceivedYet) {
+        std::cout << "Waiting for Version information to populate from system." << '\n';
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+
+    // Get the system Version struct
+    const Info::Version& systemVersion = info.get_version().second;
```
Comment (julianoes) [inline examples/system_info/system_info.cpp:88]:
```suggestion
    const Info::Version& system_version = info.get_version().second;
```

And the rest below, for consistency.

File: examples/system_info/system_info.cpp [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+              << "  flight_sw_patch: " << systemVersion.flight_sw_patch << '\n'
+              << "  flight_sw_version_type: " << systemVersion.flight_sw_version_type << '\n'
+              << "  flight_sw_vendor_major: " << systemVersion.flight_sw_vendor_major << '\n'
+              << "  flight_sw_vendor_minor: " << systemVersion.flight_sw_vendor_minor << '\n'
+              << "  flight_sw_vendor_patch: " << systemVersion.flight_sw_vendor_patch << '\n'
+              << "  flight_sw_git_hash: " << systemVersion.flight_sw_git_hash << '\n'
+              << "  os_sw_major: " << systemVersion.os_sw_major << '\n'
+              << "  os_sw_minor: " << systemVersion.os_sw_minor << '\n'
+              << "  os_sw_patch: " << systemVersion.os_sw_patch << '\n'
+              << "  os_sw_git_hash: " << systemVersion.os_sw_git_hash << '\n';
```
Comment (julianoes) [inline examples/system_info/system_info.cpp:102]: Btw. you can also just do `std::cout << systemVersion << '\n'` if that's what you want.

**Review by julianoes — COMMENTED** [review]
Some small comments, overall this is great. And make sure to update the proto submodule in the next push, then we can merge it.

**Comment by CY-1992** [comment]
@julianoes Fixed consistency issues in the example.

**Comment by julianoes** [comment]
Thanks @CY-1992. Now only the submodule update is missing. Let me try to do that real quick.

**Comment by CY-1992** [comment]
@julianoes I thought I updated the sub module with the latest commit. May I know exactly how you have updated the submodule in MAVSDK?

**Comment by julianoes** [comment]
After it has been merged in proto, I did the following steps:

```
cd proto
git fetch
git switch main
git merge
cd ..
git add proto
git commit
```

**Pull request merged**